File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/dialogue/repository/ConversationMongoRepository.java
package com.study.webflux.rag.infrastructure.dialogue.repository;

import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.study.webflux.rag.infrastructure.dialogue.adapter.persistence.document.ConversationDocument;
import reactor.core.publisher.Flux;

public interface ConversationMongoRepository
	extends
		ReactiveMongoRepository<ConversationDocument, String> {

	Flux<ConversationDocument> findBySessionIdOrderByCreatedAtDesc(String sessionId,
		Pageable pageable);

	Flux<ConversationDocument> findBySessionIdOrderByCreatedAtAsc(String sessionId,
		Pageable pageable);
}

File: backend/infrastructure/src/main/kotlin/com/miyou/app/infrastructure/dialogue/repository/ConversationMongoRepository.kt
package com.miyou.app.infrastructure.dialogue.repository

import com.miyou.app.infrastructure.dialogue.adapter.persistence.document.ConversationDocument
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface ConversationMongoRepository : ReactiveMongoRepository<ConversationDocument, String> {
    fun findBySessionIdOrderByCreatedAtDesc(
        sessionId: String,
        pageable: Pageable,
    ): Flux<ConversationDocument>

    fun findBySessionIdOrderByCreatedAtAsc(
        sessionId: String,
        pageable: Pageable,
    ): Flux<ConversationDocument>
}
